src/main/java/org/jenkinsci/plugins/workflow/steps/TimeoutStepExecution.java

```diff
@@ -230,29 +239,41 @@ public String getShortDescription() {
         }
     }
 
-    public interface ResetCallback extends Serializable {
-        void logWritten();
-    }
+    private static final class ResetTimer extends SlaveToMasterCallable<Void, RuntimeException> {
 
-    private class ResetCallbackImpl implements ResetCallback {
         private static final long serialVersionUID = 1L;
-        @Override public void logWritten() {
-            resetTimer();
+
+        private final String id;
+
+        ResetTimer(String id) {
```

Comment:
It doesn't matter that the slave provides this value from a security standpoint, right? Even if bogus ids were passed, and the agent magically guessed a UUID for another build correctly, it would just reset the activity timer, which doesn't really matter.

Reply:
Right, it could not do much damage.

Reply:
The compatibility issue was sorted out, I think.

Reply:
Yes, I manually confirmed that there are no problems with upgrading the plugin while builds are running inside of a `timeout(activity:true...)` step, so I'll mark this thread as resolved.